### DISCUSSION TITLE
How do I select all the columns from a table, plus additional column?

### DISCUSSION THREAD

**Discussion by alexeygritsenko** [catfan/Medoo#1127] [Q&A]
The thing is, I don't keep the expiration date of the licenses because it's easy to calculate based on the release date. I tried it like this
```
    static function GetLicensesByUserId($userId)
    {
        global $database;

        $licenses = $database->select('licenses', [
            "*",
            "datetime_expiration" => Medoo::raw("<datetime> + INTERVAL 3 YEAR")
        ],
        [
            "iduser" => $userId,
            "ORDER" => [
            "id" => "DESC",
            ]
        ]);

        return $licenses;
    }
```
but it gives an error `Incorrect column name`

**Comment by catfan** [comment]
No, it's disallowed for that. It's a bad idea to select all columns with `*`.

Think about it: what if `datetime_expiration` already exists in the `licenses` table? The result of `SELECT *` will include this column and `<datetime> + INTERVAL 3 YEAR as datetime_expiration` will be conflicted.

It's always recommended to provide the specific column for the database query, for better performance and maintainability.

**Reply by alexeygritsenko** [reply]
Thanks for the reply!

This seems reasonable, but is not always appropriate. For example, our customer has production tables with over 500 columns. If I list them in a query, I will not pass my boss's code review. Instead, I should be able to use `*`.

If a column from my expression and table already exists, an exception should be thrown like `Duplicate column name`

**Reply by catfan** [reply]
If managing over 500 columns or more, that will be another topic. You may find a way to maintain all the columns for the whole project.

For example, getting all the column names at first from `SELECT * FROM table LIMIT 1`.  It's super fast.

**Reply by alexeygritsenko** [reply]
ok, i'll try this as a workaround. However, i would be glad if you planned to support wildcard `*` in future versions
